mllib/src/main/scala/org/apache/spark/ml/classification/LogisticRegression.scala

```diff
@@ -803,13 +829,17 @@ private class LogisticAggregator(
     this
   }
 
-  def count: Long = totalCnt
-
-  def loss: Double = lossSum / totalCnt
+  def loss: Double = {
+    require(weightSum >= 1.0, s"The effective number of samples should be " +
```

Comment:
For your consideration, the unbiased estimate of the variance can be found [here](https://en.wikipedia.org/wiki/Weighted_arithmetic_mean#Weighted_sample_covariance). There is no requirement for sum of weights > 1.